src/lib/resources/cronjob/destination.ts

```diff
@@ -0,0 +1,32 @@
+import type { MittwaldAPIV2Client } from "@mittwald/api-client";
+
+type CronjobCreationData = Parameters<
+  MittwaldAPIV2Client["cronjob"]["createCronjob"]
+>[0]["data"];
+
+export function buildCronjobDestination(
+  url: string | undefined,
+  command: string | undefined,
+  interpreter: string | undefined,
+): CronjobCreationData["destination"] | undefined {
+  let destination: CronjobCreationData["destination"];
+
+  if (url) {
+    destination = { url };
+  } else if (command && interpreter) {
+    let destinationInterpreter;
+    if (interpreter == "bash") {
+      destinationInterpreter = "/bin/bash";
+    } else {
+      destinationInterpreter = "/usr/bin/php";
+    }
+
+    destination = {
+      interpreter: destinationInterpreter,
+      path: command as string,
+    };
+  } else {
+    return undefined;
+  }
+  return destination;
```

Comment:
Since this is a single function, you can drop the result variable, and just `return` directly. This also reduces the conditional nesting in this function quite a bit:

```suggestion
  if (url) {
    return { url };
  } 
  
  if (command && interpreter) {
    let destinationInterpreter;
    if (interpreter == "bash") {
      destinationInterpreter = "/bin/bash";
    } else {
      destinationInterpreter = "/usr/bin/php";
    }

    return {
      interpreter: destinationInterpreter,
      path: command as string,
    };
  }
  
  return undefined;
```

Could be reduced even further, if the whole `destinationInterpreter` logic would be extracted into a separate function (say, `mapInterpreterToFullPath`):

```suggestion
  if (url) {
    return { url };
  }
  
  if (command && interpreter) {
    return {
      interpreter: mapInterpreterToFullPath(interpreter),
      path: command as string,
    };
  }

  return undefined;
```